### DISCUSSION TITLE
where is docs for Drawing Text in winui?

### DISCUSSION THREAD

**Discussion by ghost1372** [microsoft/WindowsAppSDK#4051] [Q&A] [answered]
i am trying to draw some text with effect like this:

![image](https://github.com/microsoft/WindowsAppSDK/assets/9213496/95cb8d8c-ae0c-4a0e-bdbf-4db6dd15ead3)

I remember there was a documentation page for it and it had an image with purple text. But now I can't find that page.
so Does anyone know that doc page?

**Comment by castorix** [comment]
A simple way is to use a _TextBlock_
and a _LinearGradientBrush_ for the Foreground brush :

![image](https://github.com/microsoft/WindowsAppSDK/assets/22345506/9b30775e-1025-464e-8d53-b64ade0bedf5)

**Reply by ghost1372** [reply]
i know but i am looking for a way to draw this kind of text on Background, something similar to wpf, which we can use a custom brush in background: `_brush = new DrawingBrush`

**Reply by castorix** [reply]
For background,  I generally do it with a SwapChain :

![SwapChain_Gradient](https://github.com/microsoft/WindowsAppSDK/assets/22345506/11fca2fc-e403-43a4-9b69-366bc141b4f9)

But it should be doable with a [CompositionEffectBrush](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.composition.compositioneffectbrush?view=windows-app-sdk-1.4)

**Reply by ghost1372** [reply]
i am trying to make a watermark, wpf was so easy to do this:

![image](https://github.com/microsoft/WindowsAppSDK/assets/9213496/d6fd7159-6fc0-4941-9f02-22ba248c5cc1)

**Comment by castorix** [comment]
Same code with _TranslateTransform_/_RotateTransform_  (test with GDI+ in _WM_PAINT_) for a rotated text as watermark :

![SwapChain_Watermark](https://github.com/microsoft/WindowsAppSDK/assets/22345506/fd2b5fe5-5494-4452-8c82-71831ba19d96)

**Reply by ghost1372** [reply]
can you share code?

**Reply by castorix** [reply, marked as answer]
It is quick draft code atm but it is about the same as the code from _Junjie Zhu - MSFT_ in your thread in MSDN (https://learn.microsoft.com/en-us/answers/questions/1465983/how-to-port-drawingbrush-stuff-from-wpf-to-winui-3)
The only difference is that I draw it in WM_PAINT (SetWindowSubclass, ...) and I added a SwapChainPanel (like in the GitHub sample for transparency) to show the background
But it could be simpler with Composition (although I had tried without Win2D (like in [Wice](https://github.com/aelyo-softworks/Wice)) and failed...)

**Reply by ghost1372** [reply]
I finally got what I wanted with Win2D
https://github.com/winUICommunity/WinUICommunity

![](https://raw.githubusercontent.com/WinUICommunity/Resources/main/WinUICommunityDocs/Win2d/Watermark.gif)
